### PR TITLE
Update basic modal layout for creating new column types [SCI-3921]

### PR DESCRIPTION
**Pull request by mlorb**
Jira ticket: [SCI-3921](https://biosistemika.atlassian.net/browse/SCI-3921)

### What was done
_Front end for create new column type modal._

### CODE REVIEW

File: app/views/repository_columns/_manage_column_modal.html.erb [hunk 1]
```diff
@@ -17,47 +17,60 @@
...
+                </option>
+                <option data-create-url="<%= repository_create_list_path(@repository) %>" value="RepositoryListValue">
+                  <%= t('libraries.manange_modal_column.select.repository_list_value') %>
+                </option>
+                <option data-create-url="<%= repository_create_asset_path(@repository) %>" value="RepositoryAssetValue">
+                  <%= t('libraries.manange_modal_column.select.repository_asset_value') %>
+                </option>
+              </select>
+            </div>
+          </div>
```
Comment (aignatov-bio) [inline app/views/repository_columns/_manage_column_modal.html.erb]: Clean all bootstrap elements (col-sm-9, form-group, form-control and etc). Make it more clean.

File: app/assets/javascripts/repository_columns/index.js.erb [hunk 2]
```diff
@@ -139,21 +151,18 @@
...
-          $('[data-role="tagsimput-label"]').hide();
-        }
+        $('.column-type').hide();
+        $('[data-column-type="'+$(this).val()+'"]').show();
+    });
+  }
+
+  function _initTagInput() {
+    $('[data-role="tagsinput"]').tagsinput({
+      maxChars: <%= Constants::NAME_MAX_LENGTH %>,
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js.erb]: Clean all ruby code and convert to normal JS file. You can find constans here - constants.js.erb

File: app/assets/javascripts/repository_columns/index.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  function initEditCoumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    // var colRadID = '#repository_column_data_type_repositorylistvalue';
+    // var tagsInputID = '[data-role="tagsinput"]';
+    // var formID = '[data-role="manage-repository-column-form"]';
+
+    $('[data-action="edit"]').off('click').on('click', function() {
+      var editUrl = $(this).closest('li').attr('data-edit-url');
+      $.get(editUrl, function(data) {
+        $(data.html).appendTo('body').promise().done(function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Let's clean this promise. This code already sync, remove all promises

File: app/assets/javascripts/repository_columns/index.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+          $(modalID).modal('show').promise().done(function() {
+            $(modalID).on('hidden.bs.modal', function() {
+              // remove edit modal window
+              $(modalID).remove();
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Why we need remove modalID every time?

File: app/assets/javascripts/repository_columns/index.js [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+            //   var values = JSON.parse($(tagsInputID).attr('data-value'));
+            //   $('#repository_column_data_type').val('RepositoryListValue');
+            //   $(colRadID).click().promise().done(function() {
+            //     $.each(tagsInputValues, function(index, element) {
+            //       $(tagsInputID).tagsinput('add', element);
+            //     });
+            //   });
+            // }
+
+            $('[data-action="save"]').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Looks like this do nothing, comment it

File: app/assets/javascripts/repository_columns/index.js [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+                        <%= I18n.t "libraries.repository_columns.index.delete_column" %>
+                      </button>
+                    </span>
+                  </div>
+                </li>`;
+
+    // remove element if already persent
+    $('[data-id="' + column.id + '"]').remove();
+    $(html).insertBefore('.repository-columns-body ul li:first')
+      .promise()
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: remove promise

File: app/assets/javascripts/repository_columns/index.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+      .done(function() {
+        initDeleteColumnModal();
+        initEditCoumnModal();
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Why we need every time init this. Make it as delegated actions on container itself.

File: app/assets/javascripts/repository_columns/index.js [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    });
+  }
+
+  function initDeleteColumnModal() {
+    $('[data-action="destroy"]').off('click').on('click', function() {
+      var element = $(this);
+      var modalHtml = $('#deleteRepositoryColumn');
+      $.get(element.closest('li').attr('data-destroy-url'), function(data) {
+        modalHtml.find('.modal-body').html(data.html)
+          .promise()
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: clean promise

File: app/assets/javascripts/repository_columns/index.js [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+          I18n.t('libraries.repository_columns.no_permissions'), 'danger'
+        );
+      });
+    });
+  }
+
+  function removeElementFromDom(column) {
+    $('.list-group-item[data-id="' + column.id + '"]').remove();
+    if ($('.list-group-item').length === 0) {
+      location.reload();
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Why we reload page if we don't have elements?

File: app/assets/javascripts/repository_columns/index.js [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  function initNewColumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    $('[data-action="new-column-modal"]').off('click').on('click', function() {
+      var modalUrl = $(this).attr('data-modal-url');
+      $.get(modalUrl, function(data) {
+        $(data.html).appendTo('body').promise().done(function() {
+          $(modalID).modal('show').promise().done(function() {
+            $(modalID).on('hidden.bs.modal', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: remove promises

File: app/assets/javascripts/repository_columns/index.js [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+              // remove create new modal window
+              $(modalID).remove();
+              $('.modal-backdrop').remove();
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Why we delete this modal everytime?

File: app/assets/javascripts/repository_columns/index.js [hunk 12]
```diff
@@ -0,0 +1,247 @@
+/* global I18n HelperModule animateSpinner */
+/* no-restricted-globals */
+var RepositoryColumns = (function() {
+  function initColumnTypeSelector() {
+    $('[name="repository_column[data_type]"]')
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+  // }
+
+  // @TODO refactor that
+  function initEditCoumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    // var colRadID = '#repository_column_data_type_repositorylistvalue';
+    // var tagsInputID = '[data-role="tagsinput"]';
+    // var formID = '[data-role="manage-repository-column-form"]';
+
+    $('[data-action="edit"]').off('click').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+
+  function removeElementFromDom(column) {
+    $('.list-group-item[data-id="' + column.id + '"]').remove();
+    if ($('.list-group-item').length === 0) {
+      location.reload();
+    }
+  }
+
+  function initDeleteSubmitAction(modal, form) {
+    modal.find('[data-action="delete"]').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+        animateSpinner(null, false);
+      }).on('ajax:error', function(e, xhr) {
+        animateSpinner(null, false);
+        HelperModule.flashAlertMsg(xhr.responseJSON.message, 'danger');
+      });
+    });
+  }
+
+  function initDeleteColumnModal() {
+    $('[data-action="destroy"]').off('click').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+        initDeleteColumnModal();
+        initEditCoumnModal();
+      });
+    // remove 'no column' list item
+    $('[data-attr="no-columns"]').remove();
+  }
+
+  function initNewColumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    $('[data-action="new-column-modal"]').off('click').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+              $('.modal-backdrop').remove();
+            });
+
+            initColumnTypeSelector();
+            $('[data-column-type="RepositoryTextValue"]').show();
+            setTimeout(function() {
+              $('#repository_column_name').focus();
+            }, 500);
+
+            $('[data-action="save"]').on('click', function() {
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Use more uniq class or id

File: app/assets/javascripts/repository_columns/index.js [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+        $(data.html).appendTo('body').promise().done(function() {
+          $(modalID).modal('show').promise().done(function() {
+            $(modalID).on('hidden.bs.modal', function() {
+              // remove create new modal window
+              $(modalID).remove();
+              $('.modal-backdrop').remove();
+            });
+
+            initColumnTypeSelector();
+            $('[data-column-type="RepositoryTextValue"]').show();
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Why RepositoryTextValue?

File: app/assets/javascripts/repository_columns/index.js (hunk 3 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 4 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: Don't know, I will remove it.

File: app/assets/javascripts/repository_columns/index.js (hunk 5 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: it's on TO DO list

File: app/assets/javascripts/repository_columns/index.js (hunk 6 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 8 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 10 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 11 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: Removed.

File: app/assets/javascripts/repository_columns/index.js (hunk 18 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: It's selected by default.

File: app/assets/javascripts/repository_columns/index.js (hunk 9 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: To show ''no-columns" element.

File: app/assets/javascripts/repository_columns/index.js (hunk 12 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 13 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 14 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 15 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 16 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 17 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js (hunk 7 above)
Comment (mlorb) [reply app/assets/javascripts/repository_columns/index.js]: :+1:

File: app/assets/javascripts/repository_columns/index.js [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+                $(modalID).modal('hide');
+              }
+            },
+            error: function() {
+            }
+          });
+        });
+      });
+    });
+  }
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Can you simplify it split for 2 functions. 
1) initNewColumnModal
2) initNewSubmitAction
Like we have for delete

File: app/assets/javascripts/repository_columns/index.js [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+        //     $.each(tagsInputValues, function(index, element) {
+        //       $(tagsInputID).tagsinput('add', element);
+        //     });
+        //   });
+        // }
+
+        $('#new-repo-column-submit').on('click', function() {
+          // if ($('#repository_column_data_type').val() === 'RepositoryListValue') {
+          //   $('#list_items').val($(tagsInputID).val());
+          // }
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Can you simplify it split for 2 functions. 
1) initEditColumnModal
2) initEditSubmitAction
Like we have for delete

**Pull request merged**